File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        a = defaultdict(int)
        data = []
        for index, i in enumerate(nums):
            d = target - i
            if  d in a:
               data.append(index)
               data.append(a[d])
               return data
            a[i]=index
        return data
               
        

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        // nested loop big o of n^2
        // hashmap big o of n
        // key is index
        // a+b = target, target - b = a, target - a = b
        HashMap<Integer,Integer> visited = new HashMap<>();

        for ( int i=0; i<nums.length; i++) {
            int complement = target - nums[i];
            if ( visited.containsKey(complement) ) {
                return new int[]{ visited.get(complement) , i };
            }
            visited.put( nums[i], i);
        }

        return new int[]{};
        
    }
}